app/services/host_inventory_api.rb

```diff
@@ -36,7 +36,7 @@ def create_host_in_inventory
 
   def inventory_host
     @inventory_host ||= host_already_in_inventory(@id) ||
-                        host_already_in_inventory(@hostname) ||
+                        (@hostname && host_already_in_inventory(@hostname)) ||
```

Comment:
This allows skipping the hostname check if we don't know the hostname. Just pass nil to the HostInventoryApi.

Reply:
Makes sense, thanks